### DISCUSSION TITLE
Remove hashes/nonces when unsafe-inline is set

### DISCUSSION THREAD

**Discussion by Yannik** [spatie/laravel-csp#117] [Ideas]
I have a quiet complicated CSP, which is optimized for minimal permissions, but for some pages is more open due to third party library requirements (e.g. mapbox).

By default, hashes are set for inline styles, but sometimes, it it necessary to set `unsafe-inline` because the hashes cannot be precomputed.

In this case, it would be very useful to either a) automatically or b) manually remove the hash/nonce directives, because otherwise `unsafe-inline` is ignored by the client browser per the CSP spec.

@freekmurze What do you think about this?

**Comment by vimalgorasiya** [comment]
@freekmurze 

I'm aligned with @Yannik's viewpoint regarding CSP implementation in the package. I believe that either automatically removing nonces when `unsafe-inline` is specified for scripts or providing a method to remove nonce directives would significantly enhance usability and developer friendliness.

Thanks

**Comment by freekmurze** [comment]
Feel free to submit a PR 👍

**Comment by m1-5ha** [comment]
That would be amazing ! Actual example is using tinymce requiring 'unsafe-inline' for style... https://www.tiny.cloud/docs/tinymce/latest/tinymce-and-csp/

**Comment by juliangums** [comment]
PR created for this: https://github.com/spatie/laravel-csp/pull/132
